presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -229,22 +230,21 @@ public boolean isComplete()
             @Override
             public TupleDomain<ColumnHandle> getCurrentPredicate()
```

Comment:
Asking because I am not sure if the current implementation is thread-safe:
Do we assume that `getCurrentPredicate` is called from a single thread by the coordinator?

Reply:
> Asking because I am not sure if the current implementation is thread-safe:

Do you mean updating of `currentDynamicFilter`? Yes, it could be that two threads call `getCurrentPredicate()` and both update `currentDynamicFilter`. However, even if `currentDynamicFilter` was updated with less domains, in the next call it will be updated again with more domains. It's best effort, but we should always return up-to-date result.

Reply:
Is there a guarantee that there *will* be another call? Or is it possible for that to be the last call and "get stuck" with a less complete domain?

Reply:
> Is there a guarantee that there will be another call?

@martint that's what `io.prestosql.spi.connector.DynamicFilter#isComplete` is for. Connector can check if the filter it will fetch will be final.